### ISSUE TITLE
Add a rule to point out unused variables

### ISSUE THREAD

**Issue by milancermak** [crytic/amarna#62]
Amarna already lovingly warns about unused imports, function call arguments or functions themselves. Can you add a rule to warn about unused variables to its arsenal as well?

Example:
```cairo
func foo{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}() -> (val):
    let (caller) = get_caller_address()
    return (42)
end
```

In the above example, `caller` is never used. The rule would point it out.

**Comment by fcasal** [comment]
Hi @milancermak,
Amarna already finds this with the dead-store rule: there was a value saved in `caller` but never used before a return statement:
![image](https://user-images.githubusercontent.com/17139618/181777721-36c60c55-fbd4-45e6-b433-72e9ae7848fc.png)

Let me know if there are examples where you don't see this being found!

**Comment by milancermak** [comment]
Ah, you're right. I even though I saw it before but whilst doing code review on a PR, I've noticed an unused variable but no Amarna warning attached ([link](https://github.com/BibliothecaForAdventurers/realms-contracts/pull/149/files#diff-cba5e215f1826b9fb26abd65a856f9a8af5cc284fbbe5bbf42dba0c939e29db9R131) if you're curious, line 131 in `Food.cairo`). It probably has something to do with the GitHub action, because when I run amarna locally on the file, it does catch it. Sorry for the false alarm 🙏
